src/Storage/Storage/Blob/Cmdlet/NewAzureStorageContainerSasToken.cs

```diff
@@ -112,11 +112,34 @@ public NewAzureStorageContainerSasTokenCommand(IStorageBlobManagement channel)
         public override void ExecuteCmdlet()
         {
             if (String.IsNullOrEmpty(Name)) return;
+
+            // When the input context is Oauth bases, can't generate normal SAS, but UserDelegationSas
+            bool generateUserDelegationSas = false;
+            if (Channel!=null && Channel.StorageContext!= null && Channel.StorageContext.StorageAccount.Credentials.IsToken)
+            {
+                generateUserDelegationSas = true;
+                WriteVerbose("Will generate User Delegation SAS, since input Storage Context is OAuth based.");
```

Comment:
This should be included in the ShouldProcess message as well

Reply:
fixed